### DISCUSSION TITLE
How to use Secure Websocket with the swarm

### DISCUSSION THREAD

**Discussion by DeepHorizons** [libp2p/rust-libp2p#5215] [Q&A]
Im looking to enable secure websockets (wss) on my libp2p app. The example listed in the docs ([WsConfig in libp2p::websocket - Rust](https://docs.rs/libp2p/latest/libp2p/websocket/struct.WsConfig.html#examples)) give an example of building a transport, but now how to integrate it with the swarm. Ive tried using the WsConfig with the `.with_websocket()` function but I dont think I can use it there. Next I tried to use `.with_other_transport()` but could not figure out how to map the connections to what the function wants.

How can I add a wss transport to the swarm so I can listen on /ip/tcp/0/wss/%2Ftest (/ip/tcp/0/wss//test)

**Comment by drHuangMHT** [comment]
I used `with_websocket(libp2p_noise::Config::new, libp2p_yamux::Config::default)` and apparently nothing panicked when I tried to perform a Kadelima bootstrap. I'm not sure if the websocket connection actually works and it is not a secure websocket connection but I hope this helps. Noise is a framework for message encryption by the way.
